File: app/src/main/java/org/phenoapps/intercross/application/Intercross.java
package org.phenoapps.intercross.application;

import android.app.Application;

import org.phenoapps.intercross.BuildConfig;

import dagger.hilt.android.HiltAndroidApp;

@HiltAndroidApp
public class Intercross extends Application {

    public Intercross() {
        if (BuildConfig.DEBUG) {
            //StrictMode.enableDefaults();
            //un-comment to enable strict warnings in logcat
        }
    }
}

File: app/src/main/java/org/phenoapps/intercross/application/Intercross.kt
package org.phenoapps.intercross.application

import android.app.Application
import dagger.hilt.android.HiltAndroidApp
import org.phenoapps.brapi.account.BrapiAccountRepository
import javax.inject.Inject

@HiltAndroidApp
class Intercross : Application() {

    @Inject
    lateinit var brapiAccountRepository: BrapiAccountRepository

    override fun onCreate() {
        super.onCreate()
        // Run one-time migration from SharedPreferences to AccountManager.
        // Must run before any BrAPI UI or auth flow is accessed.
        brapiAccountRepository.migrateFromPrefsIfNeeded()
    }
}